### PR TITLE
Fix lint pipeline 

### PR DESCRIPTION
**Pull request by pfltdv**
#### Summary
PR contains changes to fix lint pipeline for multi-product architecture. We need to clone mattermost-server repo for the lint to work.

### CODE REVIEW

**Comment by pfltdv** [comment]
Hello all,

Do you know why we are treating CI different in go work setup.
https://github.com/mattermost/focalboard/blob/main/mattermost-plugin/build/gowork/main.go#L47-L58

**Review by spirosoik — APPROVED** [review]

File: .github/workflows/lint-server.yml [hunk 1]
```diff
@@ -16,7 +19,24 @@ jobs:
         with:
           go-version: 1.18.1
       - uses: actions/checkout@v3
+        with:
+          path: "focalboard"
+      - id: "mattermostServer"
```
Comment (pfltdv) [inline .github/workflows/lint-server.yml:24]: Checkout same branch name on mattermost-server, if missing than checkout default branch.

File: .github/workflows/lint-server.yml [hunk 2]
```diff
@@ -7,6 +7,9 @@ on:
     branches: [ main, release-** ]
   workflow_dispatch:
 
+env:
+ BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
```
Comment (pfltdv) [inline .github/workflows/lint-server.yml:11]: For prs, head_ref is the source branch, otherwise use ref_name as branch name.

**Review by agnivade — APPROVED** [review]
Thank you @pfltdv !

Some additional notes:
1. We need similar thing setup from mattermost-server side as well.
2. And for the enterprise repo as well.

File: .github/workflows/lint-server.yml [hunk 3]
```diff
@@ -16,7 +19,24 @@ jobs:
         with:
           go-version: 1.18.1
       - uses: actions/checkout@v3
+        with:
+          path: "focalboard"
+      - id: "mattermostServer"
+        uses: actions/checkout@v3
+        continue-on-error: true
+        with:
+          repository: "mattermost/mattermost-server"
```
Comment (phoinixgrr) [inline .github/workflows/lint-server.yml:28]:
```suggestion
          repository: "mattermost/mattermost-server"
          fetch-depth: "20" 
```

By default actions/checkout@v3 does a `semi-shallow clone` with depth:1 to improve cloning performance. 
Let's bump that to a reasonable higher (20) to be on the safe side.

File: .github/workflows/lint-server.yml [hunk 4]
```diff
@@ -16,7 +19,24 @@ jobs:
...
+          path: "mattermost-server"
+          ref: ${{ env.BRANCH_NAME }}
+      - uses: actions/checkout@v3
+        if: steps.mattermostServer.outcome == 'failure'
+        with:
+          repository: "mattermost/mattermost-server"
```
Comment (phoinixgrr) [inline .github/workflows/lint-server.yml:34]: Same as above: 

```suggestion
          repository: "mattermost/mattermost-server"
          fetch-depth: "20" 
```

**Review by phoinixgrr — COMMENTED** [review]
Awesome 🎖️ 

Left two minor comments

File: .github/workflows/lint-server.yml (hunk 4 above)
Comment (spirosoik) [reply .github/workflows/lint-server.yml:34]: why not just fetch latest commit?

Comment (phoinixgrr) [reply .github/workflows/lint-server.yml:35]: I have seen issues with that approach, with desired commits missing when cloned.
MergeCommits are also counted as an actual commit. 

This is an internal git clone, generated from within gitHub(gitHub runners) towards gitHub. The performance gain should be negligible

**Comment by pfltdv** [comment]
> Awesome medal_military
> 
> Left two minor comments

Changes are applied.

**Comment by pfltdv** [comment]
> Thank you @pfltdv !
> 
> Some additional notes:
> 
> 1. We need similar thing setup from mattermost-server side as well.
> 2. And for the enterprise repo as well.

I will create tickets for mattermost-server and enterprise repo.

**Review by phoinixgrr — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM

**Comment by agnivade** [comment]
@pfltdv - Can we merge this? My PR https://github.com/mattermost/focalboard/pull/3401 is blocked due to this.

**Pull request merged**